binder/src/main/java/io/grpc/binder/internal/ActiveTransportTracker.java

```diff
@@ -0,0 +1,95 @@
+package io.grpc.binder.internal;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import javax.annotation.concurrent.GuardedBy;
+import io.grpc.Attributes;
+import io.grpc.Metadata;
+import io.grpc.internal.ServerListener;
+import io.grpc.internal.ServerStream;
+import io.grpc.internal.ServerTransport;
+import io.grpc.internal.ServerTransportListener;
+
+/**
+ * Tracks which {@link BinderTransport.BinderServerTransport} are currently active and allows
+ * invoking a {@link Runnable} only once all transports are terminated.
+ */
+final class ActiveTransportTracker implements ServerListener  {
+  private final ServerListener delegate;
+  private final Runnable terminationListener;
+  @GuardedBy("this")
+  private boolean shutdown = false;
+  @GuardedBy("this")
+  private int activeTransportCount = 0;
+
+  /**
+   * @param terminationListener invoked only once the server has started shutdown
```

Comment:
Can you javadoc the other @param as well?

Reply:
Updated.